src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -250,6 +253,66 @@ LedgerMaster::addHeldTransaction (
     mHeldTransactions.insert (transaction->getSTransaction ());
 }
 
+// Validate a ledger's close time and sequence number if we're considering
+// jumping to that ledger. This helps defend agains some rare hostile or
+// insane majority scenarios.
+bool
+LedgerMaster::canBeCurrent (std::shared_ptr<Ledger const> const& ledger)
+{
+    assert (ledger);
+
+    // Never jump to a candidate ledger that precedes our
+    // last validated ledger
+
+    auto validLedger = getValidatedLedger();
+    if (validLedger &&
+        (ledger->info().seq < validLedger->info().seq))
+    {
+        JLOG (m_journal.trace()) << "Candidate for current ledger has low seq "
+            << ledger->info().seq << " < " << validLedger->info().seq;
+        return false;
+    }
+
+    // Ensure this ledger's parent close time is within five minutes of
+    // our current time. If we already have a known fully-valid ledger
+    // we perform this check. Otherwise, we only do it if we've built a
+    // few ledgers as our clock can be off when we first start up
+
+    auto closeTime = app_.timeKeeper().closeTime();
+    auto ledgerClose = ledger->info().parentCloseTime;
+    if ((validLedger || (ledger->info().seq > 10)) &&
+        ((std::max (closeTime, ledgerClose) - std::min (closeTime, ledgerClose))
```

Comment:
@HowardHinnant, is `abs` not available for `std::chrono::duration`? Both `closeTime` and `ledgerClose` are of type `time_point` so, being able to get the absolute difference between them seems reasonable?

Reply:
Oh!!!  But it looks like these `time_point`s are based on an unsigned rep and both the official C++17 `abs(duration)` and our `date::abs` preview will refuse to compile when asked to return the abs of an unsigned duration.  This is a safety precaution to find exactly this kind of bug at compile time.  `abs(lesser - greater)`  never gives a good result in unsigned arithmetic.